### PR TITLE
ZFIN-8985 remove Criteria usage in HibernateInfrastructureRepository, HibernateAntibodyRepository

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: source/org/zfin/infrastructure/repository/HibernateInfrastructureRepository.java [hunk 1]
```diff
@@ -558,34 +431,24 @@ public GenericTerm getRootTerm(String ontologyName) {
...
-        return (PublicationAttribution) criteria.uniqueResult();
-    }
-
     @Override
     public List<Updates> getUpdates(String zdbID) {
         return HibernateUtil.currentSession()
-            .createCriteria(Updates.class)
-            .add(Restrictions.eq("recID", zdbID))
-            .addOrder(Order.desc("whenUpdated"))
+            .createQuery("from Updates where recID = :recID order by whenUpdated", Updates.class)
```
Comment (rtaylorzfin) [inline source/org/zfin/infrastructure/repository/HibernateInfrastructureRepository.java:450]: order should be desc?

File: source/org/zfin/infrastructure/repository/HibernateInfrastructureRepository.java [hunk 2]
```diff
@@ -1510,11 +1256,10 @@ public List<List<String>> executeNativeQuery(DatabaseJdbcStatement statement, Se
     @Override
     public UnloadInfo getUnloadInfo() {
         Session session = HibernateUtil.currentSession();
-        Criteria criteria = session.createCriteria(UnloadInfo.class);
-        criteria.addOrder(Order.desc("date"));
-        criteria.setFirstResult(0);
-        criteria.setMaxResults(1);
-        return (UnloadInfo) criteria.uniqueResult();
+        Query<UnloadInfo> query = session.createQuery("from UnloadInfo order by date", UnloadInfo.class);
```
Comment (rtaylorzfin) [inline source/org/zfin/infrastructure/repository/HibernateInfrastructureRepository.java:1259]: order desc

**Review by rtaylorzfin — COMMENTED** [review]
The only thing I noticed was a couple places where the `order by ... desc` logic lost the `desc` part.

File: source/org/zfin/infrastructure/repository/HibernateInfrastructureRepository.java (hunk 1 above)
Comment (cmpich) [reply source/org/zfin/infrastructure/repository/HibernateInfrastructureRepository.java:450]: Great catch!

**Review by rtaylorzfin — APPROVED** [review]
Nice work!

File: source/org/zfin/database/repository/PostgresRepository.java [hunk 3]
```diff
@@ -25,62 +17,29 @@ public class PostgresRepository {
 
     public static List<PostgresSession> getSystemDatabases(DatabaseFormBean formBean) {
         Session session = HibernateUtil.currentSession();
-        Criteria criteria = session.createCriteria(PostgresSession.class);
-        if (formBean.getOrderBy() != null)
-            criteria.addOrder(Order.asc(formBean.getOrderBy()));
-        else
-            criteria.addOrder(Order.asc("dbname"));
-        return criteria.list();
+        String fromPostgresSession = "from PostgresSession order by dbname";
```
Comment (rtaylorzfin) [inline source/org/zfin/database/repository/PostgresRepository.java:20]: do we need to preserve the logic that checks the formBean.getOrderBy() ? If not, we could remove the formBean argument.

File: source/org/zfin/database/repository/PostgresRepository.java [hunk 4]
```diff
@@ -25,62 +17,29 @@ public class PostgresRepository {
...
-        TimeZone pst = TimeZone.getTimeZone("PST");
-        return new Date(date.getTime() + pst.getRawOffset());
+        return query.list();
     }
 
-
     public static List<String> getAllDbNames() {
         Session session = HibernateUtil.currentSession();
-        Criteria criteria = session.createCriteria(PostgresSession.class);
-        criteria.setProjection(Projections.distinct(Projections.property("dbname")));
```
Comment (rtaylorzfin) [inline source/org/zfin/database/repository/PostgresRepository.java:79]: I'm not familiar with projections. I guess we don't need them here?

File: source/org/zfin/database/repository/PostgresRepository.java (hunk 3 above)
Comment (cmpich) [reply source/org/zfin/database/repository/PostgresRepository.java:20]: Yes, we want to list the different databases in alphabetical order.

**Review by rtaylorzfin — APPROVED** [review]
Looks good

File: source/org/zfin/database/repository/PostgresRepository.java (hunk 4 above)
Comment (cmpich) [reply source/org/zfin/database/repository/PostgresRepository.java:79]: I'll check this out in a future sprint. It's only used on a dev tools page.

**Pull request merged**